exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -349,6 +381,73 @@ func (r *AWSManagedMachinePoolReconciler) getEC2Service(scope scope.EC2Scope) se
 	return ec2.NewService(scope)
 }
 
+func (r *AWSManagedMachinePoolReconciler) getASGService(scope cloud.ClusterScoper) services.ASGInterface {
+	if r.asgServiceFactory != nil {
+		return r.asgServiceFactory(scope)
+	}
+	return asg.NewService(scope)
+}
+
 func (r *AWSManagedMachinePoolReconciler) getReconcileService(scope scope.EC2Scope) services.MachinePoolReconcileInterface {
 	return ec2.NewService(scope)
 }
+
+func (r *AWSManagedMachinePoolReconciler) getInfraCluster(ctx context.Context, log *logger.Logger, cluster *clusterv1.Cluster, awsMachinePool *expinfrav1.AWSManagedMachinePool) (scope.EC2Scope, error) {
```

Comment:
Already mentioned in another comment – this should already be part of the scope and not re-fetched. Neither should EKS code log anything about `AWSCluster` since that's for non-EKS.